src/lib.rs

```diff
@@ -183,16 +217,23 @@ impl BitcoinD {
 
     /// Launch the bitcoind process from the given `exe` executable with given [Conf] param
     pub fn with_conf<S: AsRef<OsStr>>(exe: S, conf: &Conf) -> Result<BitcoinD, Error> {
-        let work_dir = match &conf.tmpdir {
-            Some(path) => TempDir::new_in(path),
-            None => match env::var("TEMPDIR_ROOT") {
-                Ok(env_path) => TempDir::new_in(env_path),
-                Err(_) => TempDir::new(),
-            },
-        }?;
-        debug!("work_dir: {:?}", work_dir);
-        let datadir = work_dir.path().to_path_buf();
-        let cookie_file = datadir.join(conf.network).join(".cookie");
+        let work_dir = match (&conf.tmpdir, &conf.datadir) {
+            (Some(_), Some(_)) => {
+                return Err(Error::Custom(
```

Comment:
Could you please create a specific error instead of a generic `Custom` variant?
(same text is ok in the `fmt::Debug` is ok)

Reply:
I try to avoid error variants with reason strings because they are harder to match against the specific reason downstream.
Please use a variant like `BothDirsSpecified` which formats to "tempdir and staticdir cannot be enabled at same time" as you wrote for `reason`